ecommerce/taxes/lib/taxes/services.rb

```diff
@@ -34,4 +36,34 @@ def stream_name(order_id)
       "Taxes::Order$#{order_id}"
     end
   end
-end
\ No newline at end of file
+
+  class AddAvailableVatRateHandler
+    def initialize(event_store)
+      @catalog = VatRateCatalog.new(event_store)
+      @event_store = event_store
+    end
+
+    def call(cmd)
+      raise VatRateAlreadyExists if catalog.vat_rate_available?(cmd.vat_rate.code)
+
+      event_store.publish(available_vat_rate_added_event(cmd), stream_name: stream_name(cmd))
+    end
+
+    private
+
+    attr_reader :event_store, :catalog
+
+    def available_vat_rate_added_event(cmd)
+      AvailableVatRateAdded.new(
+        data: {
+          available_vat_rate_id: cmd.available_vat_rate_id,
+          vat_rate: cmd.vat_rate
+        }
+      )
+    end
+
+    def stream_name(cmd)
+      "Taxes::AvailableVatRate$#{cmd.vat_rate.code}"
```

Comment:
I'm using `code` instead of `uuid` in the stream name. 
This helps ensure uniqueness of vat rates and simplifies some logic. 

Please let me know what you think.

Reply:
Technically it is a good choice. However I am wondering if the model is correct. But I don't have an answer yet. It is good for now.